### PR TITLE
feat: add additional checks

### PR DESCRIPTION
**Pull request by Thegaram**
## 1. Purpose or design rationale of this PR

Add additional checks.

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [X] feat: A new feature

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [X] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [X] This PR is not a breaking change
- [ ] Yes

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved validation in cryptographic operations to detect and reject invalid public key inputs, including curve points with out-of-range coordinates

* **Tests**
  * Added test suite for cryptographic operations, covering edge cases such as invalid curve point handling and malformed public key rejection

* **Chores**
  * Patch version update

### CODE REVIEW

**Review by georgehao — APPROVED** [review]

File: crypto/secp256k1/curve.go [hunk 1]
```diff
@@ -92,6 +92,10 @@ func (BitCurve *BitCurve) Params() *elliptic.CurveParams {
 
 // IsOnCurve returns true if the given (x,y) lies on the BitCurve.
 func (BitCurve *BitCurve) IsOnCurve(x, y *big.Int) bool {
+	if x.Cmp(BitCurve.P) >= 0 || y.Cmp(BitCurve.P) >= 0 {
+		return false
+	}
```
[the remark this answers is not in the document]
Comment (noel2004) [reply crypto/secp256k1/curve.go:97]: I guess there is some constraint in external which ensure the represent of curve point is positive. So Just omit this?

**Review by noel2004 — APPROVED** [review]

File: crypto/secp256k1/curve.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Thegaram) [reply crypto/secp256k1/curve.go:97]: @coderabbitai This is not exploitable: all deserialization paths use big.Int.SetBytes() which always produces non-negative values. Negative coordinates cannot arrive through any external input path.
